config/application.rb

```diff
@@ -23,6 +23,8 @@ class Application < Rails::Application
 
     # Custom directories with classes and modules you want to be autoloadable.
     config.autoload_paths += %W(#{config.root}/vendor)
+    config.autoload_paths += %W(#{config.root}/lib/avalon)
+    config.autoload_paths += %W(#{config.root}/lib/avalon/batch)
```

Comment:
Should be using `eager_load_paths` instead for production?

```  
    config.eager_load_paths << Rails.root.join('lib')
```
Relavant information: 
http://blog.arkency.com/2014/11/dont-forget-about-eager-load-when-extending-autoload/
https://stackoverflow.com/questions/19773266/confusing-about-autoload-paths-vs-eager-load-paths-in-rails-4

No idea why they are doing /vendor and /app/models/concerns? Those should be already autoloaded/eager_loaded in Rails 4+?

https://github.com/rails/rails/blob/master/railties/lib/rails/engine/configuration.rb#L35